### ISSUE TITLE
Random access file?

### ISSUE THREAD

**Issue by fiasko131** [hierynomus/smbj#349]
Hello

I use your library with Android.
For a third-party application to read a file from mine without releasing the file, but stream it, I need to implement a http sever that will be accessible at http: // localhost: 12345.

I use NanoHTTPD to do this, which gets the inputstream from the smb file.
With the protocol smb1 and jcifs my code works correctly for audio files and video files. The jcfis library has a SmbRandomAccessFile.class class that allows the streamed file via my http server to be "seekable".

Do you think that I can find an equivalent with your library, because for the moment the stream works but the stream is reset with each action on seek.

I saw that there was an option and so I used this but without success:
```` Java
Set<SMB2CreateOptions> s = new HashSet<>();
s.add(SMB2CreateOptions.FILE_RANDOM_ACCESS);
com.hierynomus.smbj.share.File remoteSmbjFile = activity.mCurrentDiskShare
                                                            .openFile(activity.mCurrentSmb2Path+"\\"+item.fileName+"."+item.getExt()
                                                                    , EnumSet.of(AccessMask.GENERIC_ALL)
                                                                    , null
                                                                    , SMB2ShareAccess.ALL
                                                                    , SMB2CreateDisposition.FILE_OPEN
                                                                    , s);
````

**Comment by neoxpert** [comment]
Have you tried to use skip() on the InputStream instance?

**Comment by fiasko131** [comment]
@neoxpert 
yes but the result is not very good. The forward and backward are much less efficient and the latency time important compared to my smb1 files.

**Comment by neoxpert** [comment]
When it comes down to performance i can't report any issues in the applications I am using smbj so far. No matter if using skip or read with offset :-/. Both operations yield the expected results.

Are you recreating the input stream with every incoming request (reading chunks?) or do you cache the inpit streams?

**Comment by fiasko131** [comment]
@neoxpert 
I use this code of principle [StreamOverHttp](https://stackoverflow.com/questions/9058135/android-serversocket-programming-with-jcifs-streaming-files)

So theoretically the inputstream should not be recreated when skiping, if I understand correctly ..

Have you tried with android?

**Comment by neoxpert** [comment]
No I have not tried it on Anroid, but I doubt that InputStreams / Sockets differ from "normal" ones. Although for now I only had some implementations of plain HttpServlets handling range requests. If I understood the code correctly in a quick overlook, it recreates a new input stream for each requested chunk unless the range request requires the whole file to be loaded at once.

I will try to reimplement it and check it for myself.

**Comment by neoxpert** [comment]
I don't know if I did it right, but for me the InputStreams were recreated with every consecutive range request. If I got it right, jcifs does some caching for itself that improves the creation of the InputStreams.

So I would strongly recommend to implement some cache mechanism in order to get rid of the whole overhead that has to be gone through when connecting to the share etc. and have a look if that helps to reduce the overall latency.
